### DISCUSSION TITLE
Environmental variables loss

### DISCUSSION THREAD

**Discussion by antliu2012** [apache/pulsar#23066] [Q&A]
Hi , all
I’ve wrote a source  connector  and a sink connector (both of them has packaged into nar file) ,
when I run these 2 connectors on pulsar standalone mode (bin/pulsar standalone), the connectors works fine , 
But when I run these 2 connectors on pulsar cluster mode ([Deploy a cluster on bare metal | Apache Pulsar](https://pulsar.apache.org/docs/3.3.x/deploy-bare-metal/#connect-to-the-running-cluster)), I found connectors get stuck.
 after tracing log, I found some environmental variables is lost (I’ve setting these Environmental variables in /etc/profile in each machine, many  variables do not work except JAVA_HOME) .
Does anyone know the cause of the problem?

**Comment by hezhangjian** [comment]
hello, @antliu2012 I think `/etc/profile` only works in bash conversation instead of running programs. Would you mind check it from `/proc/$pid/environ`

**Reply by antliu2012** [reply]
Yes, you're right. I've checked the /proc/$pid/environ file just now , these is no Environmental variables I need . 
So, how can I setting/enable  these  environmental variables when I commit function? (/bin/pulsar sink create --name sink01 --archive xxx-sink.nar ...)
BTW, how pulsar workers run functions in other machine actually? like 'ssh host01 "java -jar xxx.nar"'? can I add 'bash -c'?  :P

**Comment by hezhangjian** [comment]
`export XX=YY; java -jar xxx.jar`

**Reply by antliu2012** [reply]
你的意思是说pulsar worker是这样运行nar文件的对吗？那我怎么在``bin/pulsar source create``的时候指定环境变量呢？
You mean this is how pulsar worker runs nar files right? And how do I specify the environment variables in ``bin/pulsar source create``?
